### ISSUE TITLE
Make it easier to access parent datasets in the UI

### ISSUE THREAD

**Issue by simonbray** [galaxyproject/galaxy#11092]
If I rerun a job in the UI, the interface shows HIDs of the input datasets. This is useful, but less so if the history is very large and not at all useful if the input is from another history (I have to know which one it came from to be able to find it again). Maybe this could be made a bit easier? I don't have a perfect solution, but for example, a view button which takes you directly to the input dataset, or a rerun button for the parent job. The latter would allow the user to navigate the whole of the dataset provenance chain in a few clicks (at least in one direction).

**Comment by mvdbeek** [comment]
I guess you know about the info button, right ?
<img width="876" alt="Screenshot 2021-01-11 at 13 02 04" src="https://user-images.githubusercontent.com/6804901/104180428-62925f80-540d-11eb-9f2a-e0855d7c56fa.png">

A click on the input takes you recursively to all inputs in the provenance chain.
Does that fit the usecase / is it sufficient to navigate to that component from the tool form ? Should this be expanded somehow ?

**Comment by simonbray** [comment]
Yeah, I know about the info button, but somehow I never realised the listed input datasets were links. Thanks for the pointing that out! I think that is a good alternative to what I was asking for.

**Comment by mvdbeek** [comment]
What do you think about the side-panel navigation and maybe a arrow in the tool interface that pops out the dataset information ?

**Comment by simonbray** [comment]
> side-panel navigation

That could be nice, it would be more useful than just providing the HID in my opinion. Not sure how this would work for inputs from another history.

> a arrow in the tool interface that pops out the dataset information

Maybe just duplicate the current info button from the history, linking to the input dataset? Then the interface stays consistent and it should be immediately clear what the new button does.

I prefer the second idea (not that they're mutually exclusive).

**Comment by mvdbeek** [comment]
> That could be nice, it would be more useful than just providing the HID in my opinion. Not sure how this would work for inputs from another history.
> 
> >

The new history panel can switch histories, that shouldn't be a big problem



> Maybe just duplicate the current info button from the history, linking to the input dataset?

So when you select a dataset, you see the info button, maybe above the parameter help text?

**Comment by simonbray** [comment]
> > That could be nice, it would be more useful than just providing the HID in my opinion. Not sure how this would work for inputs from another history.
> > >
> 
> The new history panel can switch histories, that shouldn't be a big problem

I'm sure it's technically possible, but unexpected history changes might be quite annoying.

> > Maybe just duplicate the current info button from the history, linking to the input dataset?
> 
> So when you select a dataset, you see the info button, maybe above the parameter help text?

Yes - maybe limited to job reruns.
